x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/process/MlMemoryTracker.java

```diff
@@ -524,10 +537,22 @@ private void setAnomalyDetectorJobMemoryToLimit(String jobId, ActionListener<Lon
                 // during the memory refresh.
                 logger.trace("[{}] anomaly detector job deleted during ML memory update", jobId);
             } else {
-                logger.error("[" + jobId + "] failed to get anomaly detector job during ML memory update", e);
+                logIfNecessary(
+                    () -> logger.error(
+                        () -> new ParameterizedMessage("[{}] failed to get anomaly detector job during ML memory update", jobId),
+                        e
+                    )
+                );
+
             }
             memoryRequirementByAnomalyDetectorJob.remove(jobId);
             listener.onResponse(null);
         }));
     }
+
+    private void logIfNecessary(Runnable log) {
```

Comment:
I think it would be good to add a Javadoc comment here even though it's a private method, to say that the idea is not to cause log spam when a node shuts down during async operation of this class.